src/app/index.ts

```diff
@@ -179,21 +188,36 @@ module.exports = class extends Generator {
       throw new Error("Missing project properties");
     }
 
-    let bindingWrapper = "lib/binding.js";
+    let bindingWrapperTarget = "lib/binding.js";
     const genTypeScript = (this.options as any).typescript;
+    const genIntermediate = (this.options as any).intermediate;
+
+    let bindingWrapperSrc = (genIntermediate ?
+                                "lib/binding_intermediate.js" :
+                                bindingWrapperTarget);
     if (genTypeScript) {
-      bindingWrapper = "lib/binding.ts";
+      if (genIntermediate) {
+        bindingWrapperSrc = "lib/binding_intermediate.ts";
+      }
+
+      bindingWrapperTarget = "lib/binding.ts";
```

Comment:
Aren't we forgetting to get lib/binding.ts into the Src as well?

Reply:
yes- bah, what are these tests doing...I had manually tested the combinations before to confirm that the modules did the right thing and built but then simplified the logic and relied on the unit tests to do the validation- now i'm concerned :(

Reply:
Yeah, that's definitely a problem.  Here's my proposed fix for that logic: https://github.com/kfarnung/generator-napi-module/commit/03db6c4752669e90636ecc15f4e11c705b1fdc69

Reply:
oh- i see- js is ts 😛 ... it was simply copying lib/binding.js to lib/binding.ts- and so the file with the right name was there, just the wrong contents. I still dont know why the first set of tests passed locally but failed on CI but hopefully this is right now. AFAICT, it's not a cleanup issue- every test run gets a new SHA-based directory name, so one test's execution should not affect the next from a file perspective

Reply:
You're making sure to build before you test, right?

Reply:
yup